contribs/gmf/src/services/permalink.js

```diff
@@ -876,6 +928,58 @@ gmf.Permalink.prototype.updateLayerStateByVisibility_ = function(layer) {
 
 
 /**
+ * Update all properties found in the appplication state for the given layer
+ * @param  {ol.layer.Base} layer The layer
+ * @private
+ */
+gmf.Permalink.prototype.updateLayerFromState_ = function(layer) {
+  var layerName = /** @type {string} */ (layer.get('layerName'));
+  var isMerged = /** @type {boolean} */ (layer.get('isMerged'));
+  var param, stateValue, enable;
+  Object.keys(gmf.PermalinkOpenLayersLayerProperties).forEach(function(layerProp) {
+    param = this.getLayerStateParam_(layerName, isMerged, gmf.PermalinkOpenLayersLayerProperties[layerProp]);
+    stateValue = this.ngeoStateManager_.getInitialValue(param);
+    if (stateValue !== undefined) {
+      layer.set(gmf.PermalinkOpenLayersLayerProperties[layerProp], stateValue);
+    }
+  },this);
+
+  // -- Layer Visibility -- //
+  param = this.getLayerStateParam_(layerName, isMerged);
+  stateValue = this.ngeoStateManager_.getInitialValue(param);
+  if (stateValue !== undefined) {
+    //Visibility state of the layer is not the default one -> user interacted with
+    if (isMerged) {
+      //Not mixed case, we fetched layers names, if not empty: layer must be visible
+      enable = stateValue.length > 0;
+      layer.setVisible(enable);
+    } else {
+      //Mixed case, we fetched true or false
+      enable = stateValue;
+      layer.setVisible(enable);
```

Comment:
if the compiler complains about:
```
layer.setVisible(stateValue)
```
it should also complains about:
```
enable = stateValue;
layer.setVisible(enable);
```

Can you try to change the code as suggested by Pierre and report what's the error reported by the compiler?